From 898197941100d6be843a151ee45738d1c272de87 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Mon, 31 Oct 2022 17:57:04 +0000
Subject: [PATCH] WebPageBlink: keep same window type when creating additional
 windows

If an app has a specific "defaultWindowType" in appinfo.json, if should
be kept when opening an additional window.

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
Upstream-Status: Inappropriate [LuneOS specific]

 src/platform/webengine/web_page_blink.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/platform/webengine/web_page_blink.cc b/src/platform/webengine/web_page_blink.cc
index 9c55b77..f1ec9fd 100644
--- a/src/platform/webengine/web_page_blink.cc
+++ b/src/platform/webengine/web_page_blink.cc
@@ -1290,7 +1290,8 @@ WebView* WebPageBlink::CreateWindow(const std::string& newUrl, std::unique_ptr<W
 
   // Create a new webApp instance for this page
   WebAppManager *webAppMgr = WebAppManager::Instance();
-  webAppMgr->CreateWindowForAppPage(kWtCard, app_desc_, "{}", app_id_, newPage);
+  webAppMgr->CreateWindowForAppPage(webAppMgr->WindowTypeFromString(app_desc_->DefaultWindowType()), 
+                                    app_desc_, "{}", app_id_, newPage);
   
   return newPage->PageView();
 }
